File: python_calculation/prime_number_calculation.py

def sieve_of_eratosthenes(limit):
    primes = [True] * (limit + 1)
    p = 2

    while p * p <= limit:
        if primes[p]:
            for i in range(p * p, limit + 1, p):
                primes[i] = False
        p += 1

    prime_numbers = [p for p in range(2, limit) if primes[p]]

File: rust_calculation/src/prime_number_calculation.rs
pub fn sieve_of_eratosthenes(limit: usize) -> Vec<usize> {
    let mut primes = vec![true; limit + 1];
    let mut p = 2;

    while p * p <= limit {
        if primes[p] {
            let mut i = p * p;
            while i <= limit {
                primes[i] = false;
                i += p;
            }
        }
        p += 1;
    }

    (2..limit).filter(|&p| primes[p]).collect()
}

// fn main() {
//     let prime_numbers = sieve_of_eratosthenes(100);
//     println!("{:?}", prime_numbers);
// }
